src/combat.ts

```diff
@@ -130,52 +145,277 @@ export class Macro extends LibramMacro {
+    return new Macro().meatStasis(checkPassive);
+  }
+
+  startCombat(): Macro {
+    return this.tryHaveSkill($skill`Sing Along`)
+      .tryHaveSkill($skill`Curse of Weaksauce`)
+      .trySkill($skill`Pocket Crumbs`)
+      .trySkill($skill`Extract`)
+      .tryHaveItem($item`porquoise-handled sixgun`)
+      .externalIf(have($skill`Meteor Lore`), Macro.trySkill($skill`Micrometeorite`))
+      .tryHaveItem($item`Time-Spinner`)
+      .tryHaveItem($item`Rain-Doh indigo cup`)
+      .tryHaveItem($item`Rain-Doh blue balls`)
+      .externalIf(
+        haveEquipped($item`Buddy Bjorn`) || haveEquipped($item`Crown of Thrones`),
+        Macro.while_("!pastround 3 && !hppercentbelow 25", Macro.item("seal tooth"))
+      )
+      .externalIf(
+        [
+          $familiar`Cocoabo`,
+          $familiar`Feather Boa Constrictor`,
+          $familiar`Ninja Pirate Zombie Robot`,
+          $familiar`Stocking Mimic`,
+        ].some((familiar) => myFamiliar() === familiar),
+        Macro.while_("!pastround 10 && !hppercentbelow 25", Macro.item("seal tooth"))
+      )
+      .externalIf(
+        myFamiliar() === $familiar`Hobo Monkey`,
+        Macro.while_(
+          `!match "shoulder, and hands you some Meat." && !pastround 20 && !hppercentbelow 25`,
+          Macro.item("seal tooth")
+        )
+      );
+  }
+
+  static startCombat(): Macro {
+    return new Macro().startCombat();
+  }
+
+  kill(): Macro {
+    return (
+      this.externalIf(
+        myClass() === $class`Sauceror` && have($skill`Curse of Weaksauce`),
+        Macro.trySkill($skill`Curse of Weaksauce`)
+      )
+        .externalIf(
+          !(myClass() === $class`Sauceror` && have($skill`Curse of Weaksauce`)),
+          Macro.while_("!pastround 20 && !hppercentbelow 25 && !missed 1", Macro.attack())
+        )
+        // Using while_ here in case you run out of mp
+        .while_("hasskill Saucegeyser", Macro.skill($skill`Saucegeyser`))
+        .while_("hasskill Weapon of the Pastalord", Macro.skill($skill`Weapon of the Pastalord`))
+        .while_("hasskill Cannelloni Cannon", Macro.skill($skill`Cannelloni Cannon`))
+        .while_("hasskill Wave of Sauce", Macro.skill($skill`Wave of Sauce`))
+        .while_("hasskill Saucestorm", Macro.skill($skill`Saucestorm`))
+        .while_("hasskill Lunging Thrust-Smack", Macro.skill($skill`Lunging Thrust-Smack`))
+        .attack()
+        .repeat()
+    );
+  }
+
+  static kill(): Macro {
+    return new Macro().kill();
+  }
+
+  basicCombat(): Macro {
+    return this.startCombat().kill();
+  }
+
+  static basicCombat(): Macro {
+    return new Macro().basicCombat();
+  }
+
+  ghostBustin(): Macro {
+    // Only bust ghosts if you have enough stunners to prevent getting hit
+    let stunRounds = 0;
+    let classStun: Skill | null = null;
+    let extraStun: Skill | null = null;
+    if (have($item`Rain-Doh blue balls`)) stunRounds++;
+    if (get("lovebugsUnlocked")) stunRounds++;
```

Comment:
Isn't Frost Bite a stun as well? Or is that too MP expensive?

Reply:
Nvm, I forgot we want to shoot the ghosts :/

Reply:
Although, should we consider blood bubble/pill keeper?

Reply:
You can't ensure blood bubble hasn't already triggered because it could have blocked an attack before you start shooting because of losing init or from getting hit during the Sing Along round,. As for pill keeper you're probably not going to have any blocking equipment on because the free combat farming equipment is going to be used.

Changing it to use specific equipment is an option but you'd miss out on possible free random item drops from the ring etc, and if equipment was factored in it would probably just be easier to maximize elemental res and DR instead of dealing with blockers.